src/main/java/com/google/sps/servlets/Caption.java

```diff
@@ -0,0 +1,53 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.io.InputStream;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+public class Caption {
+  /*
+   * Usage: String caption = new Caption().getCaptionFromID("abcdefg");
+   * 
+   * Return null if not found.
+   */
+
+  public CaptionService captionService;
+
+  public Caption() {
+    captionService = new CaptionService();
+  }
+
+  public void setCaptionService(CaptionService captionService) {
+    this.captionService = captionService;
+  }
```

Comment:
Also not sure about good practice here but I have another hunch lol.

Being able to reset a dependency  after the object has been created seems like it could cause issues in the future. For example, another developer resets the dependency cause it works for their use case but breaks someone else who using the same object instance (of course these behavior should be tested on any file that uses this class but can't guarantee that). So instead, we would like to gear users to create their own instance for their use cases instead of resetting. This can be enforced by making CaptionService a private member and have it be passed in the constructor.

Reply:
I see you learned about InjectMock. No need for this setter anymore lol.

Reply:
Done. Field-based injection is much cleaner than the previous ones.